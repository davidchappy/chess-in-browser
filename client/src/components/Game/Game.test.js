import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import ConnectedGame, { Game } from './index';

const gameProps = {
  game: {},
  white: {},
  black: {},
  pieces: {},
  board: {},
  selectedPiece: {},
  selectableTiles: {},
  fetching: {},
  fetched: {}
}

describe('rendering Game (Snapshot Test)', () => {
  const component = renderer.create(<Game {...gameProps}/>);
  const json = component.toJSON();

  it('renders without crashing', () => {
    expect(json).toMatchSnapshot();
  });

  it('renders Header and Table', () => {
    expect(json.children[0].props['className']).toBe('chess-header small');
    expect(json.children[1].props['className']).toBe('Table');
  });
});

describe('rendering Game (mount test)', () => {
  const gameWrapper = mount(
    <Game {...gameProps}/>
  );

  it('renders Header', () => {
    const header = gameWrapper.find('.chess-header.small');
    expect(header.length).toBe(1);
  });

  it('renders Table', () => {
    const header = gameWrapper.find('.Table');
    expect(header.length).toBe(1);
  });
});



// describe('rendering App (Snapshot Test)', () => {


//   it('renders Welcome by default', () => {
//     expect(json.children[0].props['className']).toBe('Welcome');
//   });
// });

// describe('rendering App (mount test)', () => {
//   it('renders Game if server data has been fetched', () => {
//     const appWithGameWrapper = shallow(<App fetched={true} />);
//     expect(appWithGameWrapper.node.props['children']['type']['displayName']).toBe('Connect(Game)');
//   });
// });

// describe('starting Guest Game', () => {
//   const createGuestGame = jest.fn();
//   const welcomeWrapper = mount(
//     <Game createGuestGame={createGuestGame}/>
//   )

//   it('calls createGuestGame', () => {
//     const button = welcomeWrapper.find('.Game-intro button');
//     button.simulate('click');
//     expect(createGuestGame).toBeCalled();
//   });
// });