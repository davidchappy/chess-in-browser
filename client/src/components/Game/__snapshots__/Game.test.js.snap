// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering Game (Snapshot Test) renders without crashing 1`] = `
<div
  className="Game"
>
  <div
    className="chess-header small"
  >
    <img
      alt="logo"
      className="chess-logo loading small"
      src="logo.svg"
    />
    <div
      className="Messages"
    >
      <p>
        <span>
          Now Playing: 
        </span>
        
      </p>
    </div>
  </div>
  <div
    className="Table"
  >
    <div
      className="Board"
    >
      <div
        className="tiles"
      />
    </div>
  </div>
</div>
`;
